src/main/groovy/org/shipkit/gradle/notes/FetchContributorsTask.java

```diff
@@ -25,9 +26,15 @@
     @Input private String readOnlyAuthToken;
     @OutputFile private File outputFile;
 
+    private ContributorsProvider contributorsProvider;
+
+    public void setContributorsProvider(ContributorsProvider contributorsProvider) {
```

Comment:
Let's avoid adding this method to the public API because it leaks internal "ContributorsProvider" type. I scrutinize all changes to public API before 1.0 release. If you want, you can put "contributorsProvider" as a field in the "FetchContributors" class. However, I don't think you really need this. When you implement new kind of ContributorsProvider, you can just create a new task type, for example "GitLabContributorsTask" (using GitLab as an example). In the implementation of the task you can use new ContributorsProvider and just reuse serialization logic.

I'm trying to avoid making the public API leaking internal API (or making the public API expose some complicated API).

Hope it makes sense :) Kind of hard to explain in text!